### PR TITLE
RELATED: RAIL-3428 - Cleanup, sanity and fixes in CLI

### PR DESCRIPTION
**Pull request by lupko**
- backend option moved to program-level so that it's together with program
- messaging fixes
- fixed link/unlink not picking dashboard id from .env

---

Supported PR commands:

| Command                  | Description            |
| ------------------------ | ---------------------- |
| `ok to test`             | Re-run standard checks |
| `extended test`          | BackstopJS tests       |
| `extended check sonar`   | SonarQube tests        |
| `extended check cypress` | Cypress E2E tests      |

---

# PR Checklist

-   [ ] commit messages adhere to the [commit message guidelines](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#what-should-the-commits-look-like)
-   [ ] review was done by a Code owner [if necessary](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-tell-if-my-pull-request-needs-approval-by-a-code-owner) (if you think it is not necessary, explain the reasoning in the description or in a comment)
-   [ ] `check` passes
-   [ ] `check-extended` passes
-   [ ] `check-extended-cypress` passes
-   [ ] `rush change` [was run if applicable](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-describe-my-changes-for-the-changelog)

### CODE REVIEW

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/5782/ : SUCCESS in 10s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/7533/ : SUCCESS in 20m 01s

File: tools/dashboard-plugin-template/README.template.md [hunk 1]
```diff
@@ -72,13 +72,36 @@ Building a new plugin is easy. Before you start, ensure that your `.env` and `.e
...
+    Run the `{{packageManager}} run add-plugin -- "https://your.hosting/pluginDirOfYourChoice/dp_{{pluginIdentifier}}.js"` to
+    create a new dashboard plugin object in the workspace specified in the `.env` file. The created dashboard object
+    point to the URL of the built plugin.
 
-    TBD
+    After successful creation of the plugin object, the tool will print plugin object identifier. You will need this
+    identifier later to link dashboard(s) with the plugin.
+
+    Note: the CLI tool has options that allow you to add plugin to different backends and/or different workspaces. Check out
+    `{{packageManager}} gdc-plugins -- --help` to learn more about the tool's commands and options.
```
Comment (no23reason) [inline tools/dashboard-plugin-template/README.template.md:90]: Will this work with npm? I know `yarn dependency` will run it, not so sure about npm though.

**Comment by gdgate** [comment]
Build in pipeline check aborted.

File: tools/dashboard-plugin-template/README.template.md (hunk 1 above)
Comment (lupko) [reply tools/dashboard-plugin-template/README.template.md]: fixed.. i forgot to add `run` there same as it is in the other commands

**Review by no23reason — APPROVED** [review]
LGTM

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by yenkins** [comment]
### Sonar scan result


More detail, see in https://sonarqube-gate.intgdc.com/dashboard?id=gooddata-ui-sdk-gate-PR1906

To scan for vulnerabilities in dependencies and run unit tests (to get coverage report in sonar) please comment your PR with 'extended check sonar'.

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-fossa-scanning-zuul-el7/1653/ : FAILURE in 3m 08s (non-voting)
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-merge-compliance-check/1933/ : SUCCESS in 22s
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-sonarqube-scan-zuul-gate/730/ : SUCCESS in 16m 35s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/7536/ : SUCCESS in 15m 44s

**Pull request merged**